src/shared/dissect-image.c

```diff
@@ -3605,13 +3605,15 @@ int verity_dissect_and_mount(
                 if (r < 0)
                         return log_debug_errno(r, "Failed to parse image %s extension-release metadata: %m", dissected_image->image_name);
 
-                r = extension_release_validate(
+                r = extension_release_validate_internal(
                                 dissected_image->image_name,
                                 required_host_os_release_id,
                                 required_host_os_release_version_id,
                                 required_host_os_release_sysext_level,
                                 required_sysext_scope,
-                                extension_release);
+                                extension_release,
+                                "SYSEXT_LEVEL",
```

Comment:
rather than passing these variables by hand, pass the image class and derive them from there

Reply:
DissectedImage struct doesn't have an attribute ImageClass - would be too much to add one and then use for this one use case no? Could add in the next PR when dissect tool is made to work with syscfg too?

Reply:
it doesn't need to be added to the struct, just to the functions instead of passing strings manually

Reply:
hm oh yes, I think that would come anyway with the new dissect-tool functionality of the upcoming PR, so was thinking of just leaving that change to the new PR.

Reply:
please fix it here, no need to wait